docs/Architectural-Overview.md

```diff
@@ -125,7 +130,166 @@ Its a REST service which provides APIs to stores build wise test results and agg
+
+# Abbreviations and Definitions: 
+
+* Target: A target is an application deployment under test. 
+
+* Autobot: A program which sets up necessary environment and pre-requisites to run test automation pointing to a single target. 
+
+* Test Execution Framework: Loosely coupled test framework module responsible for collecting test metadata from Jira and executing optimized execution ticket provided by 
+
+Kafka. 
+
+* Test Executor framework: Same as a test execution framework. 
+
+* Dist Runner: Test execution framework module loosely couple for test framework and consumer module of execution framework.  It is responsible for collecting test metadata from Jira and creating an optimized test execution plan as per Jira Test Plans. 
+
+* Chaos and destructive testing framework: A sub framework responsible for implementing libraries to generate destructive test scenarios and chaos in Target deployments. 
+
+* Object Store Data Integrity framework: A sub framework which stores s3 client state and provides library to test versioned object Store data integrity interleaved across object store version upgrade and after object store downtime.  
+
+* HA Tests: High availability test cases which tests Consistency, Availability and Reliability of Object Store. 
+
+* Test Client: A separate VM or Container configured with Python Environment and prerequisites Linux and Python site packages to run automated test cases.      
+
+
+# Introduction: 
+
+Cortx-Test is distributed test execution framework comprising multiple test runners called autobots. These autobots runs heterogeneous tests of several types (functional, limits, scale, HA tests) on different targets in optimized manner and collect test results at central database (MongoDB). The test types supported are component or system level, functional tests, load tests, security tests, etc.  
+
+# Background: 
+
+For QA certification of an enterprise product, it is always a challenge to execute the test universe within a day especially since the test universe has hundreds of thousands of tests cases of several types including component, system, and performance tests. Job schedulers and CI tools like Jenkins solve this problem to an extent by providing functionality to executing multiple tests on different machines, however it is always hard to monitor numerous test executions and deploy and manage targets and complete certification of a build within a day. The algorithms and methods mentioned in the paper explain design and implementation of test executor framework which solves this problem with automatic test configuration management, parallel test execution, reporting and storage efficiency.       
```

Comment:
```suggestion
For QA certification of an enterprise product, it is always a challenge to execute the test universe within finite timeframe (typically a day) especially since the test universe has thousands of test cases of several types including component, system and performance tests. Job schedulers and CI tools like Jenkins solve this problem to an extent by providing functionality to execute multiple tests on different machines, however it is hard to monitor numerous test executions and deploy and manage targets and complete certification of a build within a day. The algorithms and methods mentioned in the paper explain design and implementation of test executor framework which solves this problem with automatic test configuration management, parallel test execution, reporting and storage efficiency.       
```